doc/index.md

```diff
@@ -1032,33 +1053,94 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+
+### `server.listen(options)` <!-- --> {#http.server.listen}
+
+Create a new instance of an HTTP Server by passing in table `options`:
+
+  - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
+  - `.cq` (optional): A cqueues controller to use
+  - `.ctx`: an `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: protocol family
+  - `.host`: address to bind to (required if not `.path`)
+  - `.port`: port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
+  - `.path`: path to UNIX socket (required if not `.host`)
+  - `.v6only`: allow ipv6 only (no ipv4-mapped-ipv6)
+  - `.mode`: fchmod or chmod socket after creating UNIX domain socket
+  - `.mask`: set and restore umask when binding UNIX domain socket
+  - `.max_concurrent`: Maximum number of connections to allow live at a time (default: infinity)
+  - `.onerror`: function that will be called when an error occurs (default: do nothing)
+  - `.onstream`: callback function for handling a new client request
+  - `.unlink`: unlink socket path before binding?
+  - `.reuseaddr`: turn on SO_REUSEADDR flag?
+  - `.reuseport`: turn on SO_REUSEPORT flag?
+  - `.socket`: A cqueues socket object
+  - `.tls`: `nil`: allow both tls and non-tls connections
+  -         `true`: allows tls connections only
+  -         `false`: allows non-tls connections only
+  - `.version`: the http version to allow to connect (default: any)
+
+
+#### Example {#http.server.listen-example}
 
-### `listen(options)` <!-- --> {#http.server.connect}
+```lua
+
+local app_server = http_server.listen {
+		host = 192.168.0.1;
+		port = 80;
+		onstream = function(server, stream)  
+			local request_headers = assert(stream:get_headers())
+			local request_method = request_headers:get ":method"
+
+			print(string.format('[%s] "%s %s HTTP/%g"  "%s" "%s" ',
+				os.date("%d/%b/%Y:%H:%M:%S %z"),
+				request_headers:get(":method") or "",
+				request_headers:get(":path") or "",
+				stream.connection.version,
+				request_headers:get("referer") or "-",
+				request_headers:get("user-agent") or "-"
+				))
+		end;
+}
+
+```
 
 
 ### `server:onerror(new_handler)` <!-- --> {#http.server:onerror}
 
+Replaces the current error handler function with `new_handler` and returns a reference to the old function. `new_handler` is required to have at least one parameter. The default error handler is `nil`, and can be set during instantiation through the `options` table passed to the [*server.listen(options)*](#server.listen) function.
```

Comment:
> `new_handler` is required to have at least one parameter

huh?

I think you read the implementation incorrectly: what it does is if you call `server:onerror()` with *no* parameters it will return the current error handler.

> The default error handler is `nil`

No it's not?